pkg/endpointmanager/manager.go

```diff
@@ -92,6 +93,11 @@ func Insert(ep *endpoint.Endpoint) error {
 	ep.UnconditionalRLock()
 	mutex.Lock()
 
+	// The endpoint's eventqueue can be safely started. Ensure that it is only
+	// started once it is exposed to the endpointmanager so that it will be
+	// stopped when the endpoint is removed from the endpointmanager.
+	ep.EventQueue.Run()
```

Comment:
I'm not the biggest fan of this - I'd prefer to be able to keep this as is, but, since endpoint removal and cleanup  is part of deletion from the daemon (and thus tied to the endpointmanager), we need to make sure that the goroutines for the EventQueue are only started once the endpoint is exposed via the endpointmanager so that we do not leak EventQueue goroutines if the endpoint is not inserted into the endpointmanager.

Reply:
As a potential alternative longer term, we could introduce an `EndpointTemplate` and only create the `Endpoint` once it is being inserted into the manager to make it very clear when the endpoint comes to live.